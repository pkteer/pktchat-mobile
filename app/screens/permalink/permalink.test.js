// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import Permalink from './permalink.js';

describe('Permalink', () => {
    const actions = {
        addUserToTeam: jest.fn(),
        closePermalink: jest.fn(),
        getPostsAround: jest.fn(),
        getPostThread: jest.fn(),
        getChannel: jest.fn(),
        getTeamByName: jest.fn(),
        handleSelectChannel: jest.fn(),
        handleTeamChange: jest.fn(),
        joinChannel: jest.fn(),
        selectPost: jest.fn(),
        removeUserFromTeam: jest.fn(),
    };

    const baseProps = {
        actions,
        channelId: 'channel_id',
        channelIsArchived: false,
        channelName: 'channel_name',
        channelTeamId: 'team_id',
        currentTeamId: 'current_team_id',
        currentUserId: 'current_user_id',
        focusedPostId: 'focused_post_id',
        isPermalink: true,
        myTeamMemberships: {},
        myChannelMemberships: {},
        onClose: jest.fn(),
        postIds: ['post_id_1', 'focused_post_id', 'post_id_3'],
        theme: Preferences.THEMES.denim,
        componentId: 'component-id',
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <Permalink {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();

        // match archived icon
        wrapper.setProps({channelIsArchived: true});
        expect(wrapper.instance().archivedIcon()).toMatchSnapshot();
    });

    test('should match state and call loadPosts on retry', () => {
        const wrapper = shallowWithIntl(
            <Permalink {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.instance().loadPosts = jest.fn();
        wrapper.instance().retry();
        expect(wrapper.instance().loadPosts).toHaveBeenCalledTimes(1);
    });

    test('should call handleClose on onNavigatorEvent(backPress)', () => {
        const wrapper = shallowWithIntl(
            <Permalink {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn()}}},
        );

        wrapper.instance().handleClose = jest.fn();
        wrapper.instance().navigationButtonPressed({buttonId: 'backPress'});
        expect(wrapper.instance().handleClose).toHaveBeenCalledTimes(1);
    });
});
