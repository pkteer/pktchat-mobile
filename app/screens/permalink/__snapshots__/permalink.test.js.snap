// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Permalink should match snapshot 1`] = `
<Connect(SafeArea)
  backgroundColor="transparent"
  excludeHeader={true}
  footerColor="transparent"
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 20,
      }
    }
    testID="permalink.screen"
  >
    <withAnimatable(View)
      animation="zoomIn"
      delay={0}
      direction="normal"
      duration={200}
      iterationCount={1}
      iterationDelay={0}
      onAnimationBegin={[Function]}
      onAnimationEnd={[Function]}
      onTransitionBegin={[Function]}
      onTransitionEnd={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 6,
          "flex": 1,
          "margin": 10,
          "opacity": 0,
        }
      }
      useNativeDriver={true}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderTopLeftRadius": 6,
            "borderTopRightRadius": 6,
            "flexDirection": "row",
            "height": 44,
            "paddingRight": 16,
            "width": "100%",
          }
        }
      >
        <TouchableOpacity
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "justifyContent": "center",
              "paddingLeft": 7,
              "width": 40,
            }
          }
        >
          <CompassIcon
            color="#3f4350"
            name="close"
            size={20}
          />
        </TouchableOpacity>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "paddingRight": 40,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "#3f4350",
                "fontSize": 17,
                "fontWeight": "600",
              }
            }
          >
            channel_name
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(63,67,80,0.2)",
              "height": 1,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            null,
          ]
        }
      >
        <Loading
          color="#3f4350"
          size="large"
          style={Object {}}
        />
      </View>
    </withAnimatable(View)>
  </View>
</Connect(SafeArea)>
`;

exports[`Permalink should match snapshot 2`] = `
<Text>
  <CompassIcon
    name="archive-outline"
    style={
      Array [
        Object {
          "color": "#3f4350",
          "fontSize": 16,
          "paddingRight": 20,
        },
      ]
    }
  />
   
</Text>
`;
